WalletWasabi/Crypto/ZeroKnowledge/RandomizedCommitments.cs

```diff
@@ -0,0 +1,31 @@
+using NBitcoin.Secp256k1;
+using WalletWasabi.Crypto.Groups;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge
+{
+	public class RandomizedCommitments
+	{
+		public static RandomizedCommitments RandomizeMAC(MAC mac, Scalar z, GroupElement Ma)
```

Comment:
when is it conventional to add a static method vs. another constructor? if this style is preferred then we may want to revise some things i've added ;-)

Reply:
Factory methods are much more flexible because the can express a meaning and are not restricted to return an specific type (they can return sub-types too).  Additional magic can happen inside the factory too and they can even throw exceptions while constructors should never throw, you can implement caching in a factory method or use a objects pool in case the creation of new instances is expensive among other many things.